src/pal/src/loader/module.cpp

```diff
@@ -1620,6 +1620,8 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
         shortAsciiName = FREEBSD_LIBC;
 #elif defined(__NetBSD__)
         shortAsciiName = "libc.so";
+#elif !HAVE_GNU_LIBNAMES_H
+        shortAsciiName = "libc.musl-x86_64.so";
```

Comment:
where does the name libc.musl-x86_64.so.1 come from ? in a default musl install, there's /lib/libc.so, and ld-musl-x86_64.so.1, which is a symlink to libc.so.